src/internal/_pipe.js

```diff
@@ -0,0 +1,14 @@
+var _slice = require('./_slice');
+
+
+module.exports = function _pipe(_f, g) {
+  return {
+    length: _f.length + g.length - 1,
+    value: function() {
+      var args = _slice(arguments);
```

Comment:
Any reason for this slice, seems unnecessary will end up just increasing the perf hit.

Honestly I think these slices should be implemented as `for`/`while`s because of hot they will be